rebench/executor.py

```diff
@@ -486,16 +517,19 @@ def _keep_alive(seconds):
             else:
                 msg = str(err)
             self.ui.error(msg, run_id, cmdline)
+            run_id.report_run_failed(cmdline, 0, output)
             return True
 
         if return_code == 127:
+            run_id.fail_immediately()
             msg = ("{ind}Error: Could not execute %s.\n"
                    + "{ind}{ind}The command was not found.\n"
                    + "{ind}Return code: %d\n"
                    + "{ind}{ind}%s.\n") % (
                        run_id.benchmark.suite.executor.name, return_code, output.strip())
             self.ui.error(msg, run_id, cmdline)
-            return True
+            run_id.report_run_failed(cmdline, 0, output)
```

Comment:
Instead of returning the `return_code` later here, which is a bit magic, and breaks the simple `True`/`False` set of values, could we just do something like:
`run_id.benchmark.suite.executor.mark_as_missing()`?
Or perhaps even `run_id.benchmark.suite.executor.binary_missing = True`?

Or better: `run_id.mark_binary_as_missing()` and then run_id does the other bits: `self.binary.suite.executor.binary_missing`.